pkg/querier/tripperware/roundtrip.go

```diff
@@ -145,22 +144,11 @@ func NewQueryTripperware(
 				if isQueryRange {
 					return queryrange.RoundTrip(r)
 				} else if isQuery {
-					// If the given query is not shardable, use downstream roundtripper.
 					query := r.FormValue("query")
 					// Check subquery step size.
 					if err := SubQueryStepSizeCheck(query, defaultSubQueryInterval, MaxStep); err != nil {
 						return nil, err
 					}
-
-					// If vertical sharding is not enabled for the tenant, use downstream roundtripper.
-					numShards := validation.SmallestPositiveIntPerTenant(tenantIDs, limits.QueryVerticalShardSize)
-					if numShards <= 1 {
-						return next.RoundTrip(r)
-					}
-					analysis, err := queryAnalyzer.Analyze(query)
-					if err != nil || !analysis.IsShardable() {
-						return next.RoundTrip(r)
-					}
```

Comment:
This was an issue I ran into. When there is no sharding `next.RoundTrip()` handler is used for instant query which means not using the Codec where Protobuf unmarshalling happens. And since we are not using the codec we expect the response to be in JSON. We can either:
1 - Add an exception to the querier handler to return JSON if the request is an instant query without sharding, and return Protobuf otherwise 
2 - Use the Codec for all instant query requests so we always expect Protobuf

I asked Ben about the reason for not using the Codec all the time like we do with queryRange, and he said that it is to save unnecessary marshalling and unmarshalling (JSON -> Proto struct -> JSON) if the response doesn't need to be merged with other ones. He recommended we remove that and see how the performance is affected